pkg/server/ca/manager.go

```diff
@@ -226,3 +217,24 @@ func (m *manager) prune() error {
 
 	return nil
 }
+
+func (m *manager) storeCACert(caCert *x509.Certificate, upstreamBundle []byte) error {
+	m.mtx.RLock()
+	storeReq := &datastore.Bundle{
+		TrustDomain: m.c.TrustDomain.String(),
+		CaCerts:     caCert.Raw,
+	}
+	m.mtx.RUnlock()
+
+	if m.c.UpstreamBundle {
+		storeReq.CaCerts = append(storeReq.CaCerts, upstreamBundle...)
```

Comment:
The mutex above is in place because we read TrustDomain from the active config, and I wanted to be extra careful if we ever introduce config reload or something in the future... since this modification doesn't copy any data out of the config struct, I don't think it needs a lock (this is the only place that storeReq is used)

Reply:
Oh I see, I thought the mutex was for protecting storeReq, my bad.